### ISSUE TITLE
Activating AdaptyUI fails if done the same way as docs says

### ISSUE THREAD

**Issue by nathanfallet** [adaptyteam/AdaptySDK-iOS#92]
In [the docs](https://adapty.io/docs/migration-to-adapty-sdk-v3), it's recommended to enable Adapty and AdaptyUI this way:
```swift
Adapty.activate(with: configurationBuilder) { error in
  // handle the error
}
// Only if you are going to use AdaptyUI
AdaptyUI.activate()
```

But when done this way, AdaptyUI fails to activate because it does not wait for Adapty to be activated.

```
AdaptyUI activate error: You should activate Adapty SDK before using AdaptyUI
v3.2.0, AdaptyUI/AdaptyUI+Public.swift#211
```
And only later in the log you can see Adapty activating:
```
Adapty activated successfully. [2UDtMB]
v3.2.0, Adapty/Adapty+Activate.swift#93
```

A way to fix this is writing it this way:
```swift
Adapty.activate(with: configurationBuilder) { error in
  if let error {
    // handle the error
    return
  }
  // Only if you are going to use AdaptyUI
  AdaptyUI.activate()
}
```
This way, AdaptyUI is only activated when Adapty is. (and it works)

Is this a bug or does docs need to be updated?

**Comment by yildirimatcioglu** [comment]
Same here version 3.2.1

**Comment by EgorSigolaev** [comment]
Same, version 3.2.1

**Comment by ryancstack** [comment]
I think these fail unless you sequentially call them, either through structured concurrency (async/await) or activating within the completion. I haven't tried the latter, but assume that's why it's failing to activate.

**Comment by x401om** [comment]
Hello! I apologize for not addressing this issue sooner.

Depending on the context, we suggest two ways to activate Adapty:

**1. Asynchronous context with async/await:**
```
let adaptyConfigBuilder = AdaptyConfiguration
    .builder(withAPIKey: "YOUR_API_KEY")

do {
    try await Adapty.activate(with: adaptyConfigBuilder)
    try await AdaptyUI.activate()
} catch {
    // Handle the error
}
```

**2. Synchronous context with callbacks (e.g., within application(_:didFinishLaunchingWithOptions:)):**
```
let adaptyConfigBuilder = AdaptyConfiguration
    .builder(withAPIKey: "YOUR_API_KEY")

Adapty.activate(with: adaptyConfigBuilder) { error in
    // Handle the error
}

AdaptyUI.activate { error in
    // Handle the error
}
```

There is no need to wait for `Adapty.activate` to complete before running `AdaptyUI.activate`. I have double-checked both methods using our demo apps, and everything appears to be working correctly on my end.

It would be helpful to have a demo project that reproduces the issue or, at the very least, verbose logs.

**Comment by EgorSigolaev** [comment]
> Hello! I apologize for not addressing this issue sooner.
> 
> Depending on the context, we suggest two ways to activate Adapty:
> 
> **1. Asynchronous context with async/await:**
> 
> ```
> let adaptyConfigBuilder = AdaptyConfiguration
>     .builder(withAPIKey: "YOUR_API_KEY")
> 
> do {
>     try await Adapty.activate(with: adaptyConfigBuilder)
>     try await AdaptyUI.activate()
> } catch {
>     // Handle the error
> }
> ```
> 
> **2. Synchronous context with callbacks (e.g., within application(_:didFinishLaunchingWithOptions:)):**
> 
> ```
> let adaptyConfigBuilder = AdaptyConfiguration
>     .builder(withAPIKey: "YOUR_API_KEY")
> 
> Adapty.activate(with: adaptyConfigBuilder) { error in
>     // Handle the error
> }
> 
> AdaptyUI.activate { error in
>     // Handle the error
> }
> ```
> 
> There is no need to wait for `Adapty.activate` to complete before running `AdaptyUI.activate`. I have double-checked both methods using our demo apps, and everything appears to be working correctly on my end.
> 
> It would be helpful to have a demo project that reproduces the issue or, at the very least, verbose logs.

I will try to reproduce the bug again. But that was definitely happening, I am not the only one who noticed.

**Comment by x401om** [comment]
@EgorSigolaev I’m not saying there isn’t an issue, I just wanted to highlight the recommended ways to call Adapty methods and ask for additional details to reproduce it. I appreciate your help.

**Comment by EgorSigolaev** [comment]
> @EgorSigolaev I’m not saying there isn’t an issue, I just wanted to highlight the recommended ways to call Adapty methods and ask for additional details to reproduce it. I appreciate your help.

I meant I used the approach of calling Adapty methods as you mentioned above. But it didn't work as expected. If I have any updates according the issue, I will write here.

**Comment by ryancstack** [comment]
Yes I experienced the same thing @x401om. Calling the methods synchronously and then attempting to download a paywall failed immediately as "adapty not activated"

**Comment by x401om** [comment]
Hello, this problem should be fixed in [3.5.0](https://github.com/adaptyteam/AdaptySDK-iOS/releases/tag/3.5.0)
